File: game/scripts/vscripts/modifiers/upgrades/shards/modifier_upgrade_phantom_strike_knives.lua
modifier_upgrade_phantom_strike_knives = class({})
function modifier_upgrade_phantom_strike_knives:RemoveOnDeath() return false end
function modifier_upgrade_phantom_strike_knives:IsPurgable() return false end
 
function modifier_upgrade_phantom_strike_knives:OnCreated(params)
    if IsServer() then
        self:SetStackCount(1)
    end
end

function modifier_upgrade_phantom_strike_knives:OnRefresh()
    if IsServer() then
        self:IncrementStackCount()
    end
end

File: src/vscripts/modifiers/upgrades/shards/modifier_upgrade_phantom_strike_knives.ts
import { registerModifier } from "../../../lib/dota_ts_adapter";
import { ModifierUpgrade } from "../modifier_upgrade";

@registerModifier({ customNameForI18n: "modifier_upgrade_phantom_strike_knives" })
export class ModifierUpgradePhantomStrikeKnives extends ModifierUpgrade {
    OnCreated() {
        if (IsServer()) {
            this.SetStackCount(1);
        }
    }

    OnRefresh() {
        if (IsServer()) {
            this.IncrementStackCount();
        }
    }
}
